src/psyclone/domain/lfric/kernel/common_arg.py

```diff
@@ -0,0 +1,196 @@
+        :type psyir: :py:class:`fparser.two.Fortran2003.Part_Ref` | \
+            :py:class:`fparser.two.Fortran2003.Structure_Constructor`
+        :param Optional[int] nargs: the number of expected arguments. \
+            Defaults to 4.
+        :param encoding: class in which the psyir argument should \
+            be encoded. Defaults to fparser.two.Fortran2003.Part_Ref.
+        :type encoding: Optional[ \
+            :py:class:`fparser.two.Fortran2003.Part_Ref` | \
+            :py:class:`fparser.two.Fortran2003.Structure_Constructor`]
+
+        :raises TypeError: if the psyir argument is not an fparser2 \
+            Part_Ref or Structure_Constructor object.
+        :raises ValueError: if the kernel metadata is not in \
+            the form arg_type(...).
+        :raises ValueError: if the kernel metadata does not contain \
+            the expected number of arguments (nargs).
+
+        '''
+        if not isinstance(psyir, encoding):
+            raise TypeError(
+                f"Expected kernel metadata to be encoded as a "
+                f"Fortran {encoding.__name__} object but found type "
+                f"'{type(psyir).__name__}' with value '{str(psyir)}'.")
+        if not psyir.children[0].tostr().lower() == "arg_type":
+            raise ValueError(
+                f"Expected kernel metadata to have the name "
+                f"'arg_type' and be in the form 'arg_type(...)', but found "
+                f"'{str(psyir)}'.")
+        if len(psyir.children[1].children) != nargs:
+            raise ValueError(
+                f"Expected kernel metadata to have {nargs} "
+                f"arguments, but found {len(psyir.children[1].children)} in "
+                f"'{str(psyir)}'.")
+
+    @property
+    def datatype(self):
+        '''
+        :returns: the datatype for this metadata argument.
+        :rtype: str
+        '''
+        return self._datatype
+
+    @staticmethod
+    @abstractmethod
+    def check_datatype(value):
+        '''Check that value is a valid datatype.'''
+
+    @datatype.setter
+    def datatype(self, value):
+        '''
+        :param str value: set the datatype to the \
+            specified value.
+        '''
+        self.check_datatype(value)
+        self._datatype = value
+
+    @staticmethod
+    @abstractmethod
+    def check_access(value):
+        '''Check that value is a valid value for the access descriptor.'''
+
+    @property
+    def access(self):
+        '''
+        :returns: the access descriptor for this scalar \
+            argument.
+        :rtype: str
+        '''
+        return self._access
+
+    @access.setter
+    def access(self, value):
+        '''
+        :param str value: set the access descriptor to the \
+            specified value.
+
+        '''
+        self.check_access(value)
+        self._access = value
+
+    @property
+    def form(self):
+        '''
+        :returns: the form descriptor for this scalar \
```

Comment:
This isn't a scalar argument.

Reply:
I thought I'd managed to eliminate all of those typos. Oh well.

Reply:
Have you forgotten to push some changes - this (and previous code) doesn't appear to have changed since my last review?

Reply:
In the end I think this was the only place where I didn't update where I meant to.